warp10/src/main/java/io/warp10/script/processing/image/Pmask.java

```diff
@@ -0,0 +1,59 @@
+//
+//   Copyright 2020  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script.processing.image;
+
+import io.warp10.script.NamedWarpScriptFunction;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.WarpScriptStackFunction;
+import processing.core.PImage;
+
+/**
+ * Call PImage#mask
+ */
+public class Pmask extends NamedWarpScriptFunction implements WarpScriptStackFunction {
+
+  public Pmask(String name) {
+    super(name);
+  }
+
+  @Override
+  public Object apply(WarpScriptStack stack) throws WarpScriptException {
+    Object top = stack.pop();
+
+    if (!(top instanceof PImage)) {
+      throw new WarpScriptException(getName() + " expects two PIMAGEs.");
+    }
+
+    PImage mask = (PImage) top;
+
+    top = stack.pop();
+
+    if (!(top instanceof PImage)) {
+      throw new WarpScriptException(getName() + " expects two PIMAGEs.");
+    }
+
+    PImage img = (PImage) top;
+
+    // Apply mask
+    img.mask(mask);
```

Comment:
You should catch the IllegalArgumentException which can be thrown by PImage#mask and convert it to a WarpScriptException.